lib/SILGen/SILGenPoly.cpp

```diff
@@ -2995,9 +2994,6 @@ static void buildThunkBody(SILGenFunction &SGF, SILLocation loc,
   // Reabstract the result.
   SILValue outerResult = resultPlanner.execute(innerResult);
 
-  // If we hopped to the target's executor, then we need to hop back.
-  prevExecutor.emit(SGF, loc);
-
```

Comment:
Is the reason why we no longer need this hop-back in the thunk because we're guaranteed to get off the actor after returning, since every async function now has the generic executor?

Reply:
Yeah, there's never a need to do a hop immediately before returning, and the optimizer should remove it anyway.